website/src/controllers/api-v3/user.js

```diff
@@ -31,4 +38,45 @@ api.getUser = {
   },
 };
 
+/**
+ * @api {post} /email/update
+ * @apiVersion 3.0.0
+ * @apiName EmailUpdate
+ * @apiGroup User
+ *
+ * @apiSuccess {Object} { status: 'ok' }
+ **/
+api.updateEmail = {
+  method: 'POST',
+  middlewares: [authWithHeaders(), cron],
+  url: '/email/update',
+  async handler (req, res) {    
+    let user = res.locals.user.toJSON();
+    
+    console.log("+++ +++ user._id is", user._id);
+    user = await User.findOne({ _id: user._id }).exec();
+    console.log('+++ +++ user.auth.local is', user.auth.local);
+
+    if (!user.auth.local.email) throw new PreconditionFailed(res.t('userHasNoLocalRegistration'));
+
+    req.checkBody('newEmail', res.t('newEmailRequired')).notEmpty().isEmail();
+    req.checkBody('password', res.t('missingPassword')).notEmpty();
+    let validationErrors = req.validationErrors();
+    if (validationErrors) throw validationErrors;
+ 
+    // check password
+    let temp_user = await User.findOne({ _id: user._id}, {"auth.local": 1}).exec();
```

Comment:
If you didn't call toJSON at the beginning (which is meant to produce a public version of the user document and so remove some sensitive data) all fields would be available without the need to reload the user ever